File: baekjoon/rust/bother_16208.rs
// Title: 귀찮음
// Link: https://www.acmicpc.net/problem/16208

use std::io::{stdin, stdout, BufRead, BufReader, BufWriter, Write};

macro_rules! read_to_tuple {
    ($buffer:expr, $($t:ty),+) => {{
        let mut line = String::new();
        let buffer = &mut $buffer;
        buffer.read_line(&mut line).unwrap();
        let mut it = line.trim().split_whitespace();
        (
            $(it.next().unwrap().parse::<$t>().unwrap()),+
        )
    }}
}

macro_rules! read_to_vector {
    ($buffer:expr, $t:ty) => {{
        let mut line = String::new();
        let buffer = &mut $buffer;
        buffer.read_line(&mut line).unwrap();
        line.trim()
            .split_whitespace()
            .map(|x| x.parse::<$t>().unwrap())
            .collect::<Vec<$t>>()
    }};
}

fn solution(mut sticks: Vec<i64>) -> String {
    let mut ans = 0;
    let mut len_stick: i64 = sticks.iter().sum();
    sticks.sort_unstable();

    for stick in sticks {
        len_stick -= stick;
        ans += stick * len_stick
    }

    ans.to_string()
}

fn main() {
    let mut reader = BufReader::new(stdin());
    let mut writer = BufWriter::new(stdout());

    read_to_tuple!(reader, i64);
    let sticks = read_to_vector!(reader, i64);
    writeln!(writer, "{}", solution(sticks)).unwrap();
    writer.flush().unwrap();
}

File: baekjoon/python/bother_16208.py
# Title: 귀찮음
# Link: https://www.acmicpc.net/problem/16208

import sys


sys.setrecursionlimit(10 ** 6)


read_single_int = lambda: int(sys.stdin.readline().strip())
read_list_int = lambda: list(map(int, sys.stdin.readline().strip().split(' ')))


def solution(n: int, sticks: list):
    ans = 0

    len_stick = sum(sticks)
    sticks = sorted(sticks)

    for stick in sticks:
        len_stick -= stick
        ans += stick * len_stick

    return ans


def main():
    n = read_single_int()
    sticks = read_list_int()
    print(solution(n, sticks))


if __name__ == '__main__':
    main()